iceoryx_utils/test/moduletests/test_cxx_expected.cpp

```diff
@@ -248,3 +257,51 @@ TEST_F(expected_test, ExpectedWithErrorConvertsToNullopt)
     optional<int> value = sut.to_optional();
     ASSERT_THAT(value.has_value(), Eq(false));
 }
+
+TEST_F(expected_test, AndThenUnpacksOptionalWhenNonEmptyOptionalValue)
+{
+    auto sut = expected<iox::cxx::optional<int>, float>::create_value(123);
+    MockCallables mocks{};
+    EXPECT_CALL(mocks, onSuccess).Times(1);
```

Comment:
I think you could get rid of `MockCallables` completely, reduce the code and test everything anyway.
Look here:
```cpp
bool andThenWasCalled = false;
sut.and_then([&andThenWasCalled](int& val) {
        andThenWasCalled = true;
        ASSERT_THAT(val, Eq(123));
    });
EXPECT_TRUE(andThenWasCalled);
```

You could also create an integer which is incremented every time you call the the lambda to verify that it is only called once.

Reply:
Will get it in the next PR.